source/Irrlicht/CMakeLists.txt

```diff
@@ -396,6 +395,24 @@ add_library(IRRGUIOBJ OBJECT
 	CGUITreeView.cpp
 	CGUIProfiler.cpp
 )
+target_include_directories(IRRMESHOBJ     PRIVATE ${link_includes})
+target_include_directories(IRROBJ         PRIVATE ${link_includes})
+target_include_directories(IRRPARTICLEOBJ PRIVATE ${link_includes})
+target_include_directories(IRRANIMOBJ     PRIVATE ${link_includes})
+target_include_directories(IRRVIDEOOBJ    PRIVATE ${link_includes})
+target_include_directories(IRRSWRENDEROBJ PRIVATE ${link_includes})
+target_include_directories(IRRIOOBJ       PRIVATE ${link_includes})
+target_include_directories(IRROTHEROBJ    PRIVATE ${link_includes})
+target_include_directories(IRRGUIOBJ      PRIVATE ${link_includes})
+target_link_libraries(IRRMESHOBJ     PRIVATE ${link_libs})
+target_link_libraries(IRROBJ         PRIVATE ${link_libs})
+target_link_libraries(IRRPARTICLEOBJ PRIVATE ${link_libs})
+target_link_libraries(IRRANIMOBJ     PRIVATE ${link_libs})
+target_link_libraries(IRRVIDEOOBJ    PRIVATE ${link_libs})
+target_link_libraries(IRRSWRENDEROBJ PRIVATE ${link_libs})
+target_link_libraries(IRRIOOBJ       PRIVATE ${link_libs})
+target_link_libraries(IRROTHEROBJ    PRIVATE ${link_libs})
+target_link_libraries(IRRGUIOBJ      PRIVATE ${link_libs})
```

Comment:
why are all these needed, doesn't the target already get the includes and libs?

Reply:
the IrrlichtMt target yes, but those are object-libraries, and they need the information on how they get compiled as well.

One alternative: we could change the CMakeLists some more to add the files to the `IrrlichtMt` target instead of creating object-libraries. Then we would need only one `target_link_libraries()` and `target_include_directories()` call. This would be a greater change (I think), I just wanted a small change.

I assume the object-libraries are used here to have a logical grouping of files, but that's just my guess

Reply:
apple has got a special handling with one of the object-libraries called `IRROTHEROBJ`. I can keep just this one object-library and add the remaining files directly to the `IrrlichtMt` library target

```cmake
elseif(APPLE)
	# Build all IRROTHEROBJ sources as objc++, including the .cpp's
	set_target_properties(IRROTHEROBJ PROPERTIES COMPILE_OPTIONS "-xobjective-c++")
	target_sources(IRROTHEROBJ PRIVATE
		CIrrDeviceOSX.mm
		CNSOGLManager.mm
	)
```

Reply:
I also fixed this today just now. I've been checking my email, but this PR was so recent I missed it. I don't know if NeroBurner is in the IRC channel; I was discussing the issue and my progress on it there.

Reply:
The change you made is effectively the exact same change I made, and the secondary solution you proposed was also my plan B, so I think we should keep this PR and roll with it.

Reply:
There are ways to avoid this big list.
You can write a macro that wraps `add_library(IRRWHATEVEROBJ OBJECT [...])` and automatically calls `target_include_directories` + `target_link_libraries`

> I assume the object-libraries are used here to have a logical grouping of files, but that's just my guess

Correct, this logical grouping was taken from the old Irrlicht Makefile. I don't think that should be dropped.

Reply:
I finally understood your remark about the useless `target_link_libraries()` calls and hence removed them with the following explanation

```
 Don't link dependencies to object libraries

The dependencies are libraries only, the headers are added to the build
instructions in a separate call. So the `target_link_libraries()` calls
are not needed for the object-libraries to be built.
```

Reply:
That's the way I did it in the first place and it broke the build on MinGW or something I thought. Maybe it was something else that I missed.

Reply:
It seems to build with https://github.com/NeroBurner/minetest/actions/runs/958376062, but that's using https://github.com/NeroBurner/minetest/tree/use_builtin_irrlichtmt_with_ci branch

screenshot of the GitHub Actions Pipeline with successful Mingw (only MSVC fails as expected)

![image](https://user-images.githubusercontent.com/9076163/122823893-31815980-d2e0-11eb-877f-e9e5e79dfafb.png)